lib/improver/generate_ancillaries/generate_ancillary.py

```diff
@@ -227,42 +228,55 @@ def gen_orography_masks(
         thresholds = Unit(units).convert(
             np.array(thresholds), standard_orography.units)
         coords = standard_orography.coords()
-        if key == 'land':  # regular topographical bands above land
-            lower_threshold, upper_threshold = thresholds
-            orog_band = np.ma.masked_where(
-                np.ma.logical_and(
-                    (standard_orography.data > lower_threshold),
-                    (standard_orography.data <= upper_threshold)),
-                standard_orography.data).mask.astype(int)
+
+        lower_threshold, upper_threshold = thresholds
+        orog_band = np.ma.masked_where(
+            np.ma.logical_and(
+                (standard_orography.data > lower_threshold),
+                (standard_orography.data <= upper_threshold)),
+            standard_orography.data).mask.astype(int)
+
+        if key == 'land':  # topographical bands above land only
+            if standard_landmask is None:
+                msg = ('To generate topography bands only over land, a land '
+                       'mask must be provided as input')
+                raise IOError(msg)
+
             if not isinstance(orog_band, np.ndarray):
                 orog_band = np.zeros(standard_orography.data.shape).astype(int)
+
             mask_data = self.sea_mask(standard_landmask.data, orog_band,
                                       sea_fill_value=0)
             mask_cube = _make_mask_cube(
                 mask_data, key, coords, topographic_bounds=thresholds,
                 topographic_units=standard_orography.units)
+        elif key == 'any_surface_type':
+            mask_cube = _make_mask_cube(
+                orog_band, key, coords, topographic_bounds=thresholds,
+                topographic_units=standard_orography.units)
         else:
             msg = 'Unknown threshold_dict key: {}'
             raise KeyError(msg.format(key))
         mask_cube.units = Unit('1')
         return mask_cube
 
-    def process(self, orography, landmask, thresholds_dict):
+    def process(self, orography, thresholds_dict, landmask=None):
         """Loops over the supplied orographic bands, adding a cube
            for each band to the mask cubelist.
 
         Args:
-            orography (cube):
-              orography on standard grid.
-
-            landmask (cube):
-              land mask on standard grid.
+            orography (iris.cube.Cube):
+                orography on standard grid.
 
             threshold_dict (dictionary):
-              definition of orography bands required.
+                definition of orography bands required.
+
+        Keyword Args:
+            landmask (iris.cube.Cube):
+                land mask on standard grid.
```

Comment:
Would help to reiterate default behaviour here - "If exists, sea points are masked; if None, sea points are included in the appropriate orography band"

Reply:
If exists and key is land. I've added something to this effect.